testing/tests/test_context_on.py

```diff
@@ -71,6 +70,27 @@ def test_simple_events(event_name, event_kind):
         assert isinstance(mgr.charm.observed[0], event_kind)
 
 
+@pytest.mark.parametrize(
+    "event_name, event_kind",
+    [
+        ("pre_series_upgrade", ops.PreSeriesUpgradeEvent),
+        ("post_series_upgrade", ops.PostSeriesUpgradeEvent),
+    ],
+)
+def test_simple_deprecated_events(event_name, event_kind):
+    ctx = scenario.Context(ContextCharm, meta=META, actions=ACTIONS)
+    # These look like:
+    #   ctx.run(ctx.on.pre_series_upgrade(), state)
+    with warnings.catch_warnings(
+        record=False, action="ignore", category=DeprecationWarning
+    ):
```

Comment:
Why `warnings.catch_warnings` instead of `pytest.warns` here?

Reply:
No reason, really, just that this was the first of these I did and I naturally reached for `warnings.catch_warnings` and then was reminded about `pytest.warns` later when I did the other ones because there was already some use of it. I'll switch this over: I think pytest.warns is the more appropriate choice in tests.

Reply:
Sweet, looks like it was breaking on some python versions anyway